### PR TITLE
Replace views/admin/variants/search.rabl with Api::Admin::VariantSerializer

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Part of  #4060

Converging what was in search.rabl into Api::Admin::VariantSerializer.

I fixed some rubocop issues, fixed a bug with the images code and simplified options_types and values display.

But mostly this is just converting and getting everything together, not reducing complexity.

But getting everything together makes it very clear the mess we have around variant names and unit names with 8 fields... :name, :options_text, :unit_value, :unit_description, :unit_to_display, :display_as, :display_name, :name_to_display ⚠️ 

#### What should we test?
Adding a variant to an order when: creating/editing an order and creating/editing a subscription.
Test the auto complete feature carefully (search for variants), make sure the fields for each result in the select box are correct and that the products can be added to the order and/or subscription.

#### Release notes
Changelog Category: Changed
Clean up old Spree code of search box for variants when adding variants to orders and subscriptions.

### CODE REVIEW

File: app/views/spree/admin/variants/_autocomplete.js.erb [hunk 1]
```diff
@@ -16,16 +16,13 @@
       </ul>
       <ul class='variant-data'>
         <li class='variant-sku'><strong><%= t('admin.sku') %>:</strong> {{variant.sku}}</li>
-        <li class='variant-on_hand'><strong><%= t('on_hand') %>:</strong> {{variant.on_hand}}</li>
+        {{#if variant.on_demand }}
```
Comment (sauloperez) [inline app/views/spree/admin/variants/_autocomplete.js.erb]: I have no idea what this `#if ` syntax is and the same time it's a JS file with ERB in it? :astonished:

Comment (luisramos0) [reply app/views/spree/admin/variants/_autocomplete.js.erb]: this is an if clause in erb.
this is not a "JS file with erb" this is an "erb file with JS". the output of this erb file is a JS template (script type='text/template') that is injected by this complex JS code: https://github.com/openfoodfoundation/spree/blob/2-0-4-stable/backend/app/assets/javascripts/admin/variant_autocomplete.js.erb

this should be a haml template under app/assets that angular code inject...
this PR solved this issue #3440 partially but the JS code still needs to be changed, but it's complicated, see the link above...

File: app/assets/stylesheets/admin/pages/subscription_line_items.css.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#add-line-item {
+  fieldset {
+    td.vertical_align_top {
```
Comment (mkllnk) [inline app/assets/stylesheets/admin/pages/subscription_line_items.css.scss]: This is definitely an improvement and semantic naming would be even better. Or in this case, do you even need the class? Is the `#add-line-item fieldset td` selector enough?

File: spec/serializers/api/admin/variant_serializer_spec.rb [hunk 3]
```diff
@@ -1,7 +1,30 @@
...
-  it "serializes a variant" do
+
+  it "serializes the variant name" do
     serializer = Api::Admin::VariantSerializer.new variant
+
+    expect(serializer.to_json).to match variant.name
+  end
+
+  it "serializes the variant options" do
+    serializer = Api::Admin::VariantSerializer.new variant
```
Comment (mkllnk) [inline spec/serializers/api/admin/variant_serializer_spec.rb]: What about let?

```rb
let(:serializer) { Api::Admin::VariantSerializer.new(variant) }
let(:json) { serializer.to_json }
```

File: app/assets/stylesheets/admin/pages/subscription_line_items.css.scss (hunk 2 above)
Comment (luisramos0) [reply app/assets/stylesheets/admin/pages/subscription_line_items.css.scss]: the third column with the button cannot have the class.
and I removed td in the following commit when reviewing rubocop issues.

**Comment by sauloperez** [comment]
There's a valid Code Climate comment that should be considered

**Comment by luisramos0** [comment]
ok @sauloperez, I changed to class  but kept the id as I am not sure it's not used in spree css/js

**Comment by RachL** [comment]
@luisramos0 I used the Duhat release shop on French staging and created order as manager, removing and editing quantities. 

However for subs, after creating a new subs for "Duhat release" I cannot reached the `/admin/subscriptions` page anymore. There is an infinite spinner and an error 500 in the dev tools.

I deleted a variant before starting to test subs. maybe it is related? I didn't select this variant in the dropdown though (I wanted to test if the product appeared or not in the dropdown). However I selected a variant that is linked to the same master product (Rhubarbe en vrac).

The other shops still can access their /subs page so there is something going on with the data from this shop.

Can this be linked to this PR? Otherwise I think we are good to go. I could create and edit subs on other shops.
Super admin access are shared in bitwarden.

**Comment by luisramos0** [comment]
The problem is related to a deleted variant Rhubarbe - Barquette and this variant is in a subscription. So, it's not related to this PR.
If you delete a variant that is used in a subscription, you break the subscriptions list for that shop...

**Pull request merged**